apps/api/src/billing/services/refill/refill.service.ts

```diff
@@ -53,14 +54,8 @@ export class RefillService {
    * @param userId - The ID of the user to top up the wallet for
    */
   async topUpWallet(amountUsd: number, userId: UserWalletOutput["userId"]) {
-    let userWallet = await this.userWalletRepository.findOneBy({ userId });
-    let currentLimit: number = 0;
-
-    if (userWallet) {
-      currentLimit = await this.balancesService.retrieveDeploymentLimit(userWallet);
-    } else {
-      userWallet = await this.walletInitializerService.initialize(userId);
-    }
+    const userWallet = await this.getOrCreateUserWallet(userId);
+    const currentLimit = (await this.balancesService.retrieveDeploymentLimit(userWallet)) || 0;
```

Comment:
non-blocking: it looks like this function is not expected to return a non-number value
```typescript
async retrieveDeploymentLimit(userWallet: Pick<UserWalletOutput, "address">): Promise<number> 
```
```suggestion
    const currentLimit = (await this.balancesService.retrieveDeploymentLimit(userWallet));
```

Reply:
True that, silly me, didn't mock the method where it should've been.